nomad/grapl-generator.nomad

```diff
@@ -0,0 +1,72 @@
+variable "tenant_id" {
+    type = string
+    description = "The opaque tenant id."
+}
+
+variable "generator_ingest" {
+    type = string
+    description = "The expected data ingest type. e.g. sysmon."
+}
+
+variable "generator_artifact_url" {
+    type = string
+    description = "The url of the generator artifact to run. Supported formats: http(s), git, hg, and s3."
+}
+
+variable "generator_count" {
+    type = number
+    default = 1
+}
+
+job "grapl-tenant-generator" {
+    namespace = "generator-${var.tenant_id}-${var.generator_ingest}"
+    datacenters = ["dc1"]
+
+    type = "service"
+
+    update {
+        # Automatically promotes to canaries if all canaries are healthy during an update / deployment 
+        auto_promote = true
+        # Auto reverts to the last stable job variant if the update fails 
+        auto_revert = true
+        # Spins up a "canary" instance of potentially destructive updates, validates that they are healthy, then promotes the instance to update
+        canary = 1
+        max_parallel = 1
+        # The min amount of reported "healthy" time before a instance is considered healthy and an allocation is opened up for further updates
+        min_healthy_time = "15s"
+    }
+
+    group "generator" {
+        network {
+            port "generator_receiver" { }
+        }
+
+        count = var.generator_count
+
+        task "tenant-generator" {
+            driver = "exec"
+
+            artifact {
+                source = var.generator_artifact_url
```

Comment:
What kind of artifact are we expecting here? Just a bare file (i.e., not zipped or compressed at all)? Where is it supposed to come from?

Reply:
> What kind of artifact are we expecting here? Just a bare file?

`artifact` automatically (if you don't disable it) decompresses archives (zip, tgz, bz2, and xz) so it doesn't necessarily have to be a bare file.

> Where is it supposed to come from?

We can host these in an S3 bucket as artifact supports http(s) and s3 (as well as git and mercurial).

Reply:
The way that the Pulumi code works now is that it knows the base URL for our Cloudsmith repository, and then it adds to that the artifact name and the pinned version we're using in order to get a final URL. As long as whatever is invoking this can get that URL here, we should be good.

Actually, what artifacts are these? Are they things we're currently generating?